File: appinventor/components/src/com/google/appinventor/components/runtime/errors/YailRuntimeError.java
// -*- mode: java; c-basic-offset: 2; -*-
// Copyright 2009-2011 Google, All Rights reserved
// Copyright 2011-2012 MIT, All rights reserved
// Released under the Apache License, Version 2.0
// http://www.apache.org/licenses/LICENSE-2.0

package com.google.appinventor.components.runtime.errors;

/**
 * Runtime error not fitting in any more specific category.
 *
 * @author halabelson@google.com (Hal Abelson)
 */
public class YailRuntimeError extends RuntimeError {

  private String errorType;

  public YailRuntimeError(String message, String errorType) {
    super(message);
    this.errorType = errorType;
  }

  public String getErrorType() {
    return errorType;
  }

}

File: appinventor/components-ios/src/YailRuntimeError.swift
// -*- mode: swift; swift-mode:basic-offset: 2; -*-
// Copyright 2016-2023 MIT, All rights reserved
// Released under the Apache License, Version 2.0
// http://www.apache.org/licenses/LICENSE-2.0

import Foundation

@objc open class YailRuntimeError : NSException, Error, @unchecked Sendable {
  @objc public init(_ message: String, _ errorType: String) {
    super.init(name: errorType as NSString as NSExceptionName, reason: message, userInfo: nil)
  }
  
  public required init?(coder aDecoder: NSCoder) {
    super.init(coder: aDecoder)
    showAlert(message: "init(coder:) has not been implemented")
  }
}
